python/lib/evaluations/dmod/evaluations/specification/locations.py

```diff
@@ -68,8 +68,34 @@ def apply_configuration(
         template_manager: TemplateManager,
         decoder_type: typing.Type[json.JSONDecoder] = None
     ):
-        self.from_field = configuration.get("from_field", self.from_field)
-        self.pattern = configuration.get("pattern", self.pattern)
+        if "from_field" in configuration:
```

Comment:
Similar to my comment in the `dmod.evaluations.specification.fields` module, could this be simplified by creating a new instance of `LocationSpecification` from the `configuration` parameter and mutating `self` using the new instances parameters? Im probably missing something. I feel like is something you would have done if it were plausible here.

Reply:
`LocationSpecification` is one of the few classes where this could work considering that all of its fields are optional (i.e. just about any combination is a-ok). What would that look like? Just something like:

```python
location_template = self.__class__.parse_obj(configuration)

if location_template.from_field is not None:
    # assign

# etc
```
?

Reply:
> ```python
> location_template = self.__class__.parse_obj(configuration)
> 
> if location_template.from_field is not None:
>     # assign
> 
> # etc
> ```

you could even just do this:

```python
location_template = type(self).parse_obj(configuration)
```